hollow/src/main/java/com/netflix/hollow/api/producer/HollowIncrementalProducer.java

```diff
@@ -100,49 +135,63 @@ private void removeRecords(WriteState newState) {
 
             private void markRecordsToRemove(ReadState priorState, Map<String, BitSet> recordsToRemove) {
                 HollowReadStateEngine priorStateEngine = priorState.getStateEngine();
-                
-                for(Map.Entry<String, BitSet> removalEntry : recordsToRemove.entrySet()) {
+
+                for (Map.Entry<String, BitSet> removalEntry : recordsToRemove.entrySet()) {
                     markTypeRecordsToRemove(priorStateEngine, removalEntry.getKey(), removalEntry.getValue());
                 }
-                
+
                 TransitiveSetTraverser.addTransitiveMatches(priorStateEngine, recordsToRemove);
                 TransitiveSetTraverser.removeReferencedOutsideClosure(priorStateEngine, recordsToRemove);
             }
 
-            private void markTypeRecordsToRemove(HollowReadStateEngine priorStateEngine, String type, BitSet typeRecordsToRemove) {
+            private void markTypeRecordsToRemove(HollowReadStateEngine priorStateEngine, final String type, final BitSet typeRecordsToRemove) {
                 HollowTypeReadState priorReadState = priorStateEngine.getTypeState(type);
                 HollowSchema schema = priorReadState.getSchema();
-                if(schema.getSchemaType() == SchemaType.OBJECT) {
-                    HollowPrimaryKeyIndex idx = new HollowPrimaryKeyIndex(priorStateEngine, ((HollowObjectSchema) schema).getPrimaryKey()); ///TODO: Should we scan instead?  Can we create this once and do delta updates?
-                    
-                    for(Map.Entry<RecordPrimaryKey, Object> entry : mutations.entrySet()) {
-                        if(entry.getKey().getType().equals(type)) {
+                if (schema.getSchemaType() == SchemaType.OBJECT) {
+                    final HollowPrimaryKeyIndex idx = new HollowPrimaryKeyIndex(priorStateEngine, ((HollowObjectSchema) schema).getPrimaryKey()); ///TODO: Should we scan instead?  Can we create this once and do delta updates?
```

Comment:
Avoid explicit `final` on local variables unless required for inner classes.

We *do* prefer `final` on immutable fields, however.